src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -39,10 +39,17 @@ class DriftTable extends Component {
         }
 
         let rows = [];
+        let rowData = [];
 
         for (let i = 0; i < data.facts.length; i += 1) {
-            const rowData = this.renderRowData(data.facts[i], data.systems);
-            rows.push(<tr>{ rowData }</tr>);
+            if (this.props.stateFilter === 'none' || this.props.stateFilter === undefined) {
+                rowData = this.renderRowData(data.facts[i], data.systems);
+                rows.push(<tr>{ rowData }</tr>);
+            }
+            else if (this.props.stateFilter === data.facts[i].state) {
+                rowData = this.renderRowData(data.facts[i], data.systems);
+                rows.push(<tr>{ rowData }</tr>);
```

Comment:
This is in both the if and else-if, so you can extract it to the line below the entire block.

Reply:
Do you mean the
```
rows.push(<tr>{ rowData }</tr>);
```
?

If I do that, I get the following:
![image](https://user-images.githubusercontent.com/15373136/53039166-8b419280-344c-11e9-8de1-a1dc13e81221.png)

Reply:
Oh right ,cause there is no default `else` in the if block so you would sometimes just get lots of repetition from the last if-else that triggered.